content/developer/tutorials/data_modules.rst

```diff
@@ -0,0 +1,906 @@
+      ``x_living_area`` and ``x_garden_area`` fields, as shown above
+    - Include the field in the form view of the ``x_estate.property`` model
+
+
+.. note::
+
+    Unlike in Python modules, it is not possible to define an inverse or search method for
+    computed fields.
+
+
+Related Fields
+--------------
+
+Related fields are a simplified form of computed fields that mirror the value of another field
+through a many2one relationship.
+
+**Reference**: the documentation related to this topic can be found in
+:ref:`reference/fields/related`.
+
+Related fields can be of any type (the type of the field at the other end of the relation
+traversal). They are defined as if one were adding the field directly to the model with the
+addition of a ``related`` attribute that specifies the target field on the related model
+that contains the value to be mirrored.
+
+For example, if we want to access the country of the buyer directly from the
+``x_estate.property`` model, we can add the following code to our data module:
+
+.. code-block:: xml
+
+    <odoo>
+        <!-- ...model definition from before... -->
+        <record model="ir.model.fields" id="field_real_estate_property_country_id">
+            <field name="model_id" ref="estate.model_real_estate_property" />
+            <field name="name">x_country_id</field>
+            <field name="field_description">Buyer's Country</field>
+            <field name="ttype">many2one</field>
+            <field name="relation">res.country</field>
+            <field name="related">x_partner_id.country_id</field>
+        </record>
+    </odoo>
+
+The ``related`` attribute is used to specify the target field on the related model that
+contains the value to be mirrored. This must be a dot-separated list of field names.
+
+.. note::
+
+    Like computed fields, related fields are not stored in the database by default.
+    You can force the storage of the field in the database by setting the ``store``
+    attribute to ``True``.
+
+Code and Business logic
+=======================
+
+Server Actions
+--------------
+
+In a Python module, you are free to define any method on your model. One common useage pattern
+is to add so-called "actions" methods to your model then bind these methods to buttons in the UI
+(e.g to confirm a quote, post an invoice, etc.).
+
+In a data module, you can achieve the same effect by defining
+:ref:`Server Actions <reference/actions/server>` bound to your model. Server actions represent
+pieces of logic that are run dynamically on the server. These actions can be configured manually
+in the databse directly via the
+:menuselection:`Settings --> Technical --> Actions --> Server Actions` menu and can be of different
+types; in our case, we will use the ``code`` type which allows to run any Python code in a
+sandboxed environment.
+
+This environment contains several utilities to help you interact with the Odoo database:
+
+- ``self``: the record on which the action is executed
```

Comment:
yes, i got bitten a few times in the butt by this because i use `records` by reflex but in the case of a compute it's `self`
See https://github.com/odoo/odoo/blob/18.0/odoo/addons/base/models/ir_model.py#L49